[WebGPU] Reduce usages of unsafe in Swift

tadeuzagallo opened the pull request:
#### d475dd8824a5353998281da54c246aa89e4677bb
<pre>
[WebGPU] Reduce usages of unsafe in Swift
<a href="https://bugs.webkit.org/show_bug.cgi?id=296998">https://bugs.webkit.org/show_bug.cgi?id=296998</a>
<a href="https://rdar.apple.com/157660610">rdar://157660610</a>

Reviewed by Mike Wyrzykowski.

Add annotations to reduce usages of `unsafe`.

* Source/WebGPU/WebGPU/APIConversions.h:
* Source/WebGPU/WebGPU/Buffer.h:
* Source/WebGPU/WebGPU/CommandBuffer.h:
* Source/WebGPU/WebGPU/CommandBuffer.mm:
* Source/WebGPU/WebGPU/CommandEncoder.swift:
(CommandEncoder_finish_thunk(_:descriptor:)):
(WebGPU.finish(_:)):
(WebGPU.errorValidatingCopyTextureToTexture(_:destination:copySize:)):
(WebGPU.errorValidatingCopyTextureToBuffer(_:destination:copySize:)):
(WebGPU.errorValidatingImageCopyBuffer(_:)):
(WebGPU.errorValidatingCopyBufferToTexture(_:destination:copySize:)):
(WebGPU.errorValidatingRenderPassDescriptor(_:)):
(WebGPU.errorValidatingTimestampWrites(_:)):
(WebGPU.errorValidatingComputePassDescriptor(_:)):
(WebGPU.beginRenderPass(_:)):
(WebGPU.copyTextureToBuffer(_:destination:copySize:)):
(WebGPU.copyBufferToTexture(_:destination:copySize:)):
(WebGPU.copyTextureToTexture(_:destination:copySize:)):
(WebGPU.beginComputePass(_:)):
* Source/WebGPU/WebGPU/ComputePassEncoder.h:
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
* Source/WebGPU/WebGPU/QuerySet.h:
* Source/WebGPU/WebGPU/RenderPassEncoder.h:
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/WebGPU.h:
(wgpuGetCommandBufferDescriptorNextChainedStruct):
(wgpuGetComputePassDescriptorNextChainedStruct):
(wgpuGetComputePassDescriptorTimestampWrites):
(wgpuGetRenderPassDescriptorNextChainedStruct):
(wgpuGetRenderPassDescriptorTimestampWrites):

Canonical link: <a href="https://commits.webkit.org/299114@main">https://commits.webkit.org/299114@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e9c5a1637db9087bdbc3ba643b0e305098c4ce64

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117323 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36993 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27609 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123422 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69309 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2b284b5b-41c4-429e-b59d-a50dd54a231d) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119198 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37689 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45580 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89038 "Found 14 new test failures: imported/w3c/web-platform-tests/css/css-text-decor/text-shadow/basic-opacity.html imported/w3c/web-platform-tests/css/css-text-decor/text-shadow/basic.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-last-word-001.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-character.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vs-float-clearance-002.html imported/w3c/web-platform-tests/css/css-transforms/animation/transform-non-invertible-discrete-interpolation.html imported/w3c/web-platform-tests/css/css-transforms/transform-box/cssbox-content-box-002.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-left-of-viewport-offscreen-new.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-on-top-of-viewport-offscreen-new.html imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-with-classes-exit.html ... (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/43707 "Found 7 new test failures: imported/w3c/web-platform-tests/wasm/core/simd/simd_align.wast.js.html imported/w3c/web-platform-tests/wasm/serialization/module/window-serviceworker-failure.https.html js/dom/JSON-parse-complex.html storage/indexeddb/modern/index-rename-1-private.html webgl/2.0.0/conformance2/textures/image_data/tex-3d-rgb5_a1-rgba-unsigned_byte.html webgl/2.0.0/conformance2/textures/webgl_canvas/tex-2d-rg32f-rg-float.html webgl/2.0.y/conformance/extensions/ext-disjoint-timer-query.html (failure)") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/dd6fad4b-1f54-4532-9044-e5a6e712b112) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120256 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30004 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105201 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69545 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9dffb3e9-2574-4697-8a78-a2632494c2db) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29061 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23314 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67095 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/109428 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99405 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23498 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126543 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/115830 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44220 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33230 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97709 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44577 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101436 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97504 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24911 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42858 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20792 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/40570 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44093 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/49752 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/144530 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43549 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37203 "Found 1 new JSC binary failure: testapi, Found 20050 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_indexOf.js.default, ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default, ChakraCore.yaml/ChakraCore/test/Function/FuncBody.bug227901.js.default, ChakraCore.yaml/ChakraCore/test/Function/apply.js.default, ChakraCore.yaml/ChakraCore/test/Function/apply3.js.default ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46894 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45245 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8612bd87e927c233dd39a1cfc591a32da3ed10ec. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49009

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0874029017c20521a2c82c6d67ebcec487e5dc78. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49009

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0874029017c20521a2c82c6d67ebcec487e5dc78. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49009

mwyrzykowski commented on Source/WebGPU/WebGPU/Internal/WebGPUSwiftInternal.h:
```diff
@@ -70,10 +71,10 @@ class Range;
 namespace WebGPU_Internal {
 
-inline void logString(const char * input)
+inline void logString(String input)
```
Should we pass by some reference or does it not matter?

tadeuzagallo committed:
[WebGPU] Reduce usages of unsafe in Swift
https://bugs.webkit.org/show_bug.cgi?id=296998
rdar://157660610

Reviewed by Mike Wyrzykowski.

Add annotations to reduce usages of `unsafe`.

* Source/WebGPU/WebGPU/APIConversions.h:
* Source/WebGPU/WebGPU/Buffer.h:
* Source/WebGPU/WebGPU/CommandBuffer.h:
* Source/WebGPU/WebGPU/CommandBuffer.mm:
* Source/WebGPU/WebGPU/CommandEncoder.swift:
(CommandEncoder_finish_thunk(_:descriptor:)):
(WebGPU.finish(_:)):
(WebGPU.errorValidatingCopyTextureToTexture(_:destination:copySize:)):
(WebGPU.errorValidatingCopyTextureToBuffer(_:destination:copySize:)):
(WebGPU.errorValidatingImageCopyBuffer(_:)):
(WebGPU.errorValidatingCopyBufferToTexture(_:destination:copySize:)):
(WebGPU.errorValidatingRenderPassDescriptor(_:)):
(WebGPU.errorValidatingTimestampWrites(_:)):
(WebGPU.errorValidatingComputePassDescriptor(_:)):
(WebGPU.beginRenderPass(_:)):
(WebGPU.copyTextureToBuffer(_:destination:copySize:)):
(WebGPU.copyBufferToTexture(_:destination:copySize:)):
(WebGPU.copyTextureToTexture(_:destination:copySize:)):
(WebGPU.beginComputePass(_:)):
* Source/WebGPU/WebGPU/ComputePassEncoder.h:
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
* Source/WebGPU/WebGPU/QuerySet.h:
* Source/WebGPU/WebGPU/RenderPassEncoder.h:
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/WebGPU.h:
(wgpuGetCommandBufferDescriptorNextChainedStruct):
(wgpuGetComputePassDescriptorNextChainedStruct):
(wgpuGetComputePassDescriptorTimestampWrites):
(wgpuGetRenderPassDescriptorNextChainedStruct):
(wgpuGetRenderPassDescriptorTimestampWrites):

Canonical link: https://commits.webkit.org/299114@main

webkit-commit-queue commented:
Committed 299114@main (d475dd8824a5): <https://commits.webkit.org/299114@main>

Reviewed commits have been landed. Closing PR #49009 and removing active labels.
